packages/pages/src/components/image/image.tsx

```diff
@@ -0,0 +1,271 @@
+import * as React from "react";
+import { useEffect, useRef, useState } from "react";
+import { ImageProps, ImageLayout } from "./types";
+
+const MKTGCDN_URL_REGEX = /(https?:\/\/a.mktgcdn.com\/p\/)(?<uuid>.+)\/(.*)/;
+
+/**
+ * Renders an image based from the Yext Knowledge Graph. Example of using the component to render
+ * simple and complex image fields from Yext Knowledge Graph:
+ * ```
+ * import { Image } from "@yext/pages/components";
+ *
+ * const simpleImage = (<Image image={document.logo} />);
+ * const complexImage = (<Image image={document.photoGallery[0]} />);
+ * ```
+ *
+ * @public
+ */
+export const Image = ({
+  image,
+  className,
+  width,
+  height,
+  aspectRatio,
+  layout = ImageLayout.INTRINSIC,
+  placeholder,
+  imgOverrides,
+  style = {},
+}: ImageProps) => {
+  const imgRef = useRef<HTMLImageElement>(null);
+  const [imgLoaded, setImgLoaded] = useState(false);
+
+  useEffect(() => {
+    if (imgRef.current?.complete) {
+      setImgLoaded(true);
+    }
+  }, []);
+
+  validateRequiredProps(
+    layout,
+    image.image.width,
+    image.image.height,
+    width,
+    height,
+    aspectRatio
+  );
+
+  const imgWidth = Math.abs(image.image.width);
+  const imgHeight = Math.abs(image.image.height);
+  const imgUUID = getImageUUID(image.image.url);
+
+  // The image is invalid, only try to load the placeholder
+  if (!imgUUID) {
+    return <>{placeholder != null && placeholder}</>;
+  }
+
+  const absWidth = width && width > 0 ? width : undefined;
+  const absHeight = height && height > 0 ? height : undefined;
+
+  const { src, imgStyle, widths } = handleLayout(
+    layout,
+    imgWidth,
+    imgHeight,
+    imgUUID,
+    style,
+    absWidth,
+    absHeight,
+    aspectRatio
+  );
+
+  // Generate Image Sourceset
+  const srcSet: string = widths
+    .map((w) => `${getImageUrl(imgUUID, w, (imgHeight / imgWidth) * w)} ${w}w`)
+    .join(", ");
+
+  return (
+    <>
+      {!imgLoaded && placeholder != null && placeholder}
+      <img
+        ref={imgRef}
+        style={imgStyle}
+        src={src}
+        className={className}
+        width={absWidth}
+        height={absHeight}
+        srcSet={srcSet}
+        loading={"lazy"}
+        onLoad={() => setImgLoaded(true)}
```

Comment:
apparently load doesn't trigger when the image is loaded in chrome so I think just checking complete is fine.

Reply:
nvm this isn't true (just tested)